mteb/tasks/Classification/pol/PolHateClassification.py

```diff
@@ -0,0 +1,83 @@
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="""
+        @inproceedings{rottger-etal-2022-multilingual,
+            title = "Multilingual {H}ate{C}heck: Functional Tests for Multilingual Hate Speech Detection Models",
+            author = {R{\"o}ttger, Paul  and
+            Seelawi, Haitham  and
+            Nozza, Debora  and
+            Talat, Zeerak  and
+            Vidgen, Bertie},
+            editor = "Narang, Kanika  and
+            Mostafazadeh Davani, Aida  and
+            Mathias, Lambert  and
+            Vidgen, Bertie  and
+            Talat, Zeerak",
+            booktitle = "Proceedings of the Sixth Workshop on Online Abuse and Harms (WOAH)",
+            month = jul,
+            year = "2022",
+            address = "Seattle, Washington (Hybrid)",
+            publisher = "Association for Computational Linguistics",
+            url = "https://aclanthology.org/2022.woah-1.15",
+            doi = "10.18653/v1/2022.woah-1.15",
+            pages = "154--169",
+            abstract = "Hate speech detection models are typically evaluated on held-out test sets. However, this risks painting an incomplete and potentially misleading picture of model performance because of increasingly well-documented systematic gaps and biases in hate speech datasets. To enable more targeted diagnostic insights, recent research has thus introduced functional tests for hate speech detection models. However, these tests currently only exist for English-language content, which means that they cannot support the development of more effective models in other languages spoken by billions across the world. To help address this issue, we introduce Multilingual HateCheck (MHC), a suite of functional tests for multilingual hate speech detection models. MHC covers 34 functionalities across ten languages, which is more languages than any other hate speech dataset. To illustrate MHC{'}s utility, we train and test a high-performing multilingual hate speech detection model, and reveal critical model weaknesses for monolingual and cross-lingual applications.",
+        }
+        """,
+        n_samples={"test": 1907},
+        avg_character_length={"test": 49.7},
+    )
+
+    def dataset_transform(self):
+        keep_cols = ["test_case", "label_gold"]
+        rename_dict = dict(zip(keep_cols, ["text", "label"]))
+        remove_cols = [
+            col for col in self.dataset["test"].column_names if col not in keep_cols
+        ]
+        self.dataset = self.dataset.rename_columns(rename_dict)
+        self.dataset = self.dataset.class_encode_column("label")
+        self.dataset = self.dataset.class_encode_column("functionality")
+        self.dataset = self.dataset["test"].train_test_split(
+            test_size=0.5, seed=42, stratify_by_column="functionality"
+        )  # balanced sampling across types of hate speech
+        self.dataset = self.dataset.remove_columns(remove_cols)
```

Comment:
It's good to remove unused columns, but I'm not sure it's useful, since the evaluator will only look at `text` and `label` columns.

Reply:
I'm gonna leave it then casue as you say it's a good practice to clean up the unrelevant columns.